vgl/pyvgl.cxx

```diff
@@ -708,6 +710,47 @@ void wrap_oriented_box_2d(py::module &m, std::string const& class_name)
     ;
 }
 
+
+template<typename T>
+void wrap_vgl_line_3d_2_points(py::module &m, std::string const& class_name) {
+  bool (*line_3d_2_points_ideal)(const vgl_line_3d_2_points<T>&, T) = &is_ideal;
+  bool (*line_3d_2_points_collinear)(const vgl_line_3d_2_points<T>&, const vgl_point_3d<T>&) = &collinear;
+  bool (*line_3d_2_points_coplanar)(const vgl_line_3d_2_points<T>&,const vgl_line_3d_2_points<T>&) = &coplanar;
+  bool (*line_3d_2_points_concurrent)(const vgl_line_3d_2_points<T>&,const vgl_line_3d_2_points<T>&) = &concurrent;
+  bool (*line_3d_2_points_coplanar_2_points)(const vgl_line_3d_2_points<T>&, const vgl_point_3d<T>&, const vgl_point_3d<T>&) = &coplanar;
+  bool (*line_3d_2_points_coplanar_3_lines)(const vgl_line_3d_2_points<T>&,const vgl_line_3d_2_points<T>&,const vgl_line_3d_2_points<T>&) = &coplanar;
+  bool (*line_3d_2_points_concurrent_3_lines)(const vgl_line_3d_2_points<T>&,const vgl_line_3d_2_points<T>&,const vgl_line_3d_2_points<T>&) = &concurrent;
```

Comment:
Why do these bool functions need to be defined separately as opposed to below in the typical manner?

Reply:
These functions are class methods, not instance methods. Since they're defined outside the class scope the regular binding definition doesn't seem to work.

This manner of definition copies the pattern from vgl_vector_3d here:
https://github.com/VisionSystemsInc/pyvxl/blob/a2b5ed324477a1f0e344ac19e7d840fb62c718eb/vgl/pyvgl.cxx#L354

Reply:
So I think this might be a little weird - you're changing this non-class function to a class method.  So I think the python code might end up like this
``` 
line1 = vgl.line_3d_2_points(stuff)
line2 = vgl.line_3d_2_points(stuff)
point = vgl.point_3d(stuff)
line1.collinear(line2, point)
```
Notice you need to have initialized an object like line1, but then can check the collinearity of line2/point.  Is that the way this would work?

Reply:
Based on my understanding of the VXL code, these are actually just overloaded regular functions.  So maybe something like this would be more appropriate (a function on `vgl`, not sure if I'm getting the `overload_cast_` statement correct)
```
m.def("collinear", overload_cast_<vgl_line_3d_2_points<T> const&, vgl_point_3d<T> const&>()(&vgl_line_3d_2_points<T>::collinear)
```

Reply:
I updated the code to do something like this. So the call should be something like vgl.collinear(some_line, some_point) on the python side

Reply:
Please use `overload_cast_` instead of `py::overload_cast`.

Use of `py::overload_cast<...>` is only allowable for C++14 compilers
https://pybind11.readthedocs.io/en/stable/classes.html#overloaded-methods:~:text=If%20you%20have%20a%20C%2B%2B14%20compatible%20compiler%202%2C%20you%20can%20use%20an%20alternative%20syntax%20to%20cast%20the%20overloaded%20function%3A

And in my understanding, VXL is still C++11 project.  That's why we use the `overload_cast_` function from `pyvxl_util.h`
https://github.com/VisionSystemsInc/pyvxl/blob/a2b5ed324477a1f0e344ac19e7d840fb62c718eb/pyvxl_util.h#L11-L14